lib/matplotlib/_type1font.py

```diff
@@ -768,13 +861,262 @@ def transform(self, effects):
                     )
                 len0 += len(value) - pos1 + pos0
 
-        data = bytes(data)
-        return Type1Font((
-            data[:len0],
-            self._encrypt(data[len0:], 'eexec'),
+        return data[:len0], data[len0:]
+
+    def subset(self, characters: Iterable[int]) -> T.Self:
+        """
+        Return a new font that only defines the given characters.
+
+        Parameters
+        ----------
+        characters : sequence of bytes
+            The subset of characters to include
+
+        Returns
+        -------
+        `Type1Font`
+        """
+        characters = frozenset(characters)
+        encoding = {code: glyph
+                    for code, glyph in self.prop['Encoding'].items()
+                    if code in characters}
+        encoding[0] = '.notdef'
+        # todo and done include strings (glyph names)
+        todo = set(encoding.values())
+        done: set[int] = set()
+        seen_subrs = {0, 1, 2, 3}
+        while todo - done:
+            glyph = next(iter(todo - done))
+            called_glyphs, called_subrs, _, _ = self._simulate(glyph, [], [])
+            todo.update(called_glyphs)
```

Comment:
`todo.update(called_glyphs - done)` here and then the loop can just be `while todo: glyph = todo.pop(); ...`?